### PR TITLE
DEVCON-3342DemoPageIntlCalculations

### PR DESCRIPTION
**Pull request by WayneMyerAvalara**
Add logic to Demo Page to include Cross Border calculations.

### CODE REVIEW

**Comment by WayneMyerAvalara** [comment]
Only the JSON code works right now. There are some cases where the HS Code doesn't provide a duty, although the messaging will say that _de minimis_ has been met. I would like to sit with Jeff Meisner and see how he feels about that before writing out the other language logic.

File: public/js/site.js [hunk 1]
```diff
@@ -31,44 +33,95 @@ function getCompareDate() {
...
+                lines += hsCode.length ? `
+            hsCode: "${hsCode}"` : "" 
+                lines += (mass.length && unit.length) ? "," : "";
+                lines += (mass.length && unit.length) ?
+                `   
+            parameters: {
+                Mass: "${mass}",
+                Mass.UOM: "${unit}"
+            }` : "";                    
+                lines += `
```
Comment (contygm) [inline public/js/site.js]: This line looks like it's leftover

File: public/js/site.js [hunk 2]
```diff
@@ -31,44 +33,95 @@ function getCompareDate() {
...
+                    lines += hsCode.length ? `
+            'hsCode': '${hsCode}'` : "" 
+            lines += (mass.length && unit.length) ? "," : "";
+            lines += (mass.length && unit.length) ?
+            `   
+            'parameters': {
+                'Mass': '${mass}',
+                'Mass.UOM': '${unit}'
+            }` : "";                    
+        lines += `
```
Comment (contygm) [inline public/js/site.js]: this also looks like it was leftover

File: public/js/site.js [hunk 3]
```diff
@@ -77,16 +130,34 @@ function lineBuilder(reqType) {
...
+                    lines += (mass.length && unit.length) ? "," : "";
+                    lines += (mass.length && unit.length) ?
+                `   
+            parameters: {
+                Mass: "${mass}",
+                Mass.UOM: "${unit}"
+            }` : "";                                    
+                if (lineNum !== allProducts.length) lines += ',\n        ';
+                    
+                lines += `
```
Comment (contygm) [inline public/js/site.js]: this also looks like it was leftover

File: public/js/site.js [hunk 4]
```diff
@@ -31,44 +33,95 @@ function getCompareDate() {
...
+                break;           
+            case 'Ruby':            
                 lines += `{
             amount: "${amount}",
             description: "${description}",
             number: "${lineNum}",
-            taxCode: "${taxCode}"
+            taxCode: "${taxCode}"`;
+                lines += hsCode.length ? "," : "";
+                lines += hsCode.length ? `
```
Comment (contygm) [inline public/js/site.js]: looks like an unfinished duplicate of the line above.

File: public/js/site.js [hunk 5]
```diff
@@ -31,44 +33,95 @@ function getCompareDate() {
...
+            hsCode: "${hsCode}"` : "" 
+                lines += (mass.length && unit.length) ? "," : "";
+                lines += (mass.length && unit.length) ?
```
Comment (contygm) [inline public/js/site.js]: looks like an unfinished duplicate of the line above.

File: public/js/site.js [hunk 6]
```diff
@@ -31,44 +33,95 @@ function getCompareDate() {
...
                 break;
             case 'Python':
                 lines += `{
             'amount': '${amount}',
             'description': '${description}',
             'number': '${lineNum}',
-            'taxCode': '${taxCode}'
+            'taxCode': '${taxCode}'`;
+                    lines += hsCode.length ? "," : "";
+                    lines += hsCode.length ? `
```
Comment (contygm) [inline public/js/site.js]: looks like an unfinished duplicate of the line above.

File: public/js/site.js [hunk 7]
```diff
@@ -31,44 +33,95 @@ function getCompareDate() {
...
+            'hsCode': '${hsCode}'` : "" 
+            lines += (mass.length && unit.length) ? "," : "";
+            lines += (mass.length && unit.length) ?
```
Comment (contygm) [inline public/js/site.js]: looks like an unfinished duplicate of the line above.

File: public/js/site.js [hunk 8]
```diff
@@ -77,16 +130,34 @@ function lineBuilder(reqType) {
             number = ${lineNum},
             quantity = 1,
             amount = ${amount},
-            taxCode = "${taxCode}"
+            taxCode = "${taxCode}"`;
+                    lines += hsCode.length ? "," : "";
+                    lines += hsCode.length ? `
```
Comment (contygm) [inline public/js/site.js]: looks like an unfinished duplicate of the line above.

File: public/js/site.js [hunk 9]
```diff
@@ -77,16 +130,34 @@ function lineBuilder(reqType) {
...
+            hsCode: "${hsCode}"` : "" 
+                    lines += (mass.length && unit.length) ? "," : "";
+                    lines += (mass.length && unit.length) ?
```
Comment (contygm) [inline public/js/site.js]: looks like an unfinished duplicate of the line above.

File: public/js/site.js [hunk 10]
```diff
@@ -99,6 +170,10 @@ function lineBuilder(reqType) {
     return lines;
 }
 
+function lanaugeSpecificLineBuilder(indent, indented, aWrap, vWrap, assignment, objToTranslate) {
```
Comment (contygm) [inline public/js/site.js]: leftover code

File: public/js/site.js [hunk 11]
```diff
@@ -158,6 +233,9 @@ function addressBuilder(reqType, addressName, prefix) {
 // HELPER: check if shipFrom address is selected
 function shipFromChecked() {
```
Comment (contygm) [inline public/js/site.js]: maybe update the name of this func since it also sets the `isIntlTransaction` var now

File: public/js/site.js [hunk 12]
```diff
@@ -423,12 +507,14 @@ function javascriptSampleData() {
...
-    machineName: "your-machine-name"
-};
+    let config = {
+        appName: "your-app",
+        appVersion: "1.0",
+        environment: "sandbox",
+        machineName: "your-machine-name"
+    };
+
+    const sampleData = `const config = ${JSON.stringify(config, null, 4)};
```
Comment (contygm) [inline public/js/site.js]: looks like you need a closing `

File: public/js/site.js [hunk 13]
```diff
@@ -440,14 +526,13 @@ var client = new Avatax(config).withSecurity(creds);
...
+    date: ${today.toISOString().split('T')[0]},
+    customerCode: "ABC", ${isIntlTransaction ? `\n    isSellerImporterOfRecord: "true",`:``}
     addresses: {
         ${address}
     },
-    lines: [
-        ${lines}
-    ],
+    lines: 
+        ${JSON.stringify(lines, null, 4)}
```
Comment (contygm) [inline public/js/site.js]: you can put this on the same line with the `lines:` key

File: public/js/site.js (hunk 1 above)
Comment (WayneMyerAvalara) [reply public/js/site.js]: That's not a leftover. It is spaced to provide formatting in the code example. That's also why the indentation in that switch statement is all over the place.

File: public/js/site.js (hunk 2 above)
Comment (WayneMyerAvalara) [reply public/js/site.js]: Also not a leftover. Code example formatting.

File: public/js/site.js (hunk 4 above)
Comment (WayneMyerAvalara) [reply public/js/site.js]: Still not leftovers. Not even Thanksgiving stuffing made into waffles leftovers.

File: public/js/site.js (hunk 5 above)
Comment (WayneMyerAvalara) [reply public/js/site.js]: Formatting of code examples. Into code waffles.

File: public/js/site.js (hunk 6 above)
Comment (WayneMyerAvalara) [reply public/js/site.js]: Waffles.

File: public/js/site.js (hunk 7 above)
Comment (WayneMyerAvalara) [reply public/js/site.js]: Formatted waffles.

File: public/js/site.js (hunk 8 above)
Comment (WayneMyerAvalara) [reply public/js/site.js]: With butter and maple syrup.

File: public/js/site.js (hunk 9 above)
Comment (WayneMyerAvalara) [reply public/js/site.js]: Eggs over easy and bacon on the side. All waffles. All formatting. All the time.

File: public/js/site.js (hunk 10 above)
Comment (WayneMyerAvalara) [reply public/js/site.js]: Yeah, now that's some stale leftovers that need disposing.

**Comment by WayneMyerAvalara** [comment]
VAT/IOR was added to the DEMOPAGE company by Aaron Robles. That's why UK was providing a Zero Tax calculation. The Cross Border logic is still relevant and different from VAT.

**Pull request merged**